pkg/recipes/driver/bicep.go

```diff
@@ -140,18 +140,19 @@ func (d *bicepDriver) Execute(ctx context.Context, opts ExecuteOptions) (*recipe
 		deploymentID.String(),
 		clients.DeploymentsClientAPIVersion,
 	)
+
 	if err != nil {
-		return nil, recipes.NewRecipeError(recipes.RecipeDeploymentFailed, err.Error(), recipes_util.ExecutionError, recipes.GetRecipeErrorDetails(err))
+		return nil, recipes.NewRecipeError(recipes.RecipeDeploymentFailed, fmt.Sprintf("failed to deploy recipe %s of type %s", opts.BaseOptions.Recipe.Name, opts.BaseOptions.Definition.ResourceType), recipes_util.ExecutionError, recipes.GetErrorDetails(err))
 	}
```

Comment:
Do we need to be update the `message` field from `err.Error()` for rest of the references to NewRecipeError as well?

Reply:
Actually looking at NewRecipeError again I take this back. With how NewRecipeError is implemented, it is a bit confusing on when to use the original error in the message vs when it will be wrapped and populated under details, but going to leave that out for now and not increase the scope of this PR.